library/settingloader.lua

```diff
@@ -70,9 +74,22 @@ end
 function SettingLoader.load(filename)
     local f = io.open(filename, "rb")
     if f == nil then
-        return {}
+        return {
+            ["startup"] = {},
+            ["runtime-global"] = {},
+            ["runtime-per-user"] = {}
+        }
+    end
+    local version = {
+        major = SettingLoader.readUshort(f),
+        minor = SettingLoader.readUshort(f),
+        patch = SettingLoader.readUshort(f),
+        map = SettingLoader.readUshort(f)
+    }
+    if (version.major == 0 and version.minor >= 17) or version.major >= 1 then
+        -- read extranous byte present since 0.17
+        readAll(f, 1)
```

Comment:
I don't know what this byte is used for, but I will try to find out. For what the calculator needs it seems to be irrelevant though.

Reply:
Got confirmation from Bilka that this byte indeed has no meaning, see the [updated documentation](https://wiki.factorio.com/Mod_settings_file_format) on the wiki.